src/common/libutil/fluid.c

```diff
@@ -169,27 +170,68 @@ static int b58revenc (char *buf, int bufsz, fluid_t id)
     return index;
 }
 
+static inline bool is_utf8_locale (void)
+{
+    /* Assume if MB_CUR_MAX > 1, this locale can handle wide characters
+     *  and therefore will properly handle UTF-8.
+     */
+    return (MB_CUR_MAX > 1);
+}
+
+static int locale_is_utf8 (void)
```

Comment:
Good thought! Honestly I had attempted this fix so many different ways that I think I lost sight of the forest for the trees.

> if we end up using wide characters in other places besides fluid_f58_encode() are we going to regret burying the locale initialization in the fluid class? 

Yeah, from what I know, it does seem a bit wrong to call `setlocale(3)` down in the depths of a library call. However, `setlocale(LC_CTYPE, "")` should be idempotent afaik. This would, however, step on any program that explicitly chooses a different locale than the environment or system defaults dictate.

> Would it be slightly more straightforward to just document that users of fluid_f58_encode() should be sure to call setlocale(LC_TYPE, "") in their main program initialization?

I wouldn't call it more straightforward, since we'd have to figure out where to document the instructions (`fluid_f58_encode()` is not a public function), and the documentation has to be added to both the C and Python bindings. That being said, it seems like the right thing to do.

Maybe Python already initializes the locale for all I know.

I'll work on a fix for this. It will make the changes in `libutil/fluid.c` trivial at least.

Thanks!

Reply:
[This](https://docs.python.org/3/library/locale.html#background-details-hints-tips-and-caveats) section of the Python `locale` documentation backs up what you are saying @garlick:

> It is generally a bad idea to call setlocale() in some library routine, since as a side effect it affects the entire program. Saving and restoring it is almost as bad: it is expensive and affects other threads that happen to run before the settings have been restored.

I am very glad you spoke up in this PR.

It also sounds like with Python at least, `LC_CTYPE` is automatically set from the environment by default, so no need to initialize locale when using Python bindings (which is the majority of use cases currently)

```
$ python3 -c 'import locale; print (locale.getlocale())'
('en_US', 'UTF-8')
```

Reply:
Hey great, so `flux jobs` should just work once `fluid_f58_encode()` respects `MB_CUR_MAX`?

Reply:
Yes. That looks to be the case.